index.js

```diff
@@ -42,6 +46,7 @@ module.exports = {
     "max-line-length": [ 80, {
       ignore: "non-comments",
     } ],
+    "max-nesting-depth": 3,
```

Comment:
I would justify it like this: level 1 is all natural. Level 2 is for pseudo-elements, and some rare cases when one needs to style, i.e., `.Foo-bar` inside `.Foo--fancy`. Level 3 is for `.Foo--fancy .Foo-bar`'s pseudo-elements, pseudo-classes and some very special cases. And more levels than that means a proplem.

Many people and teams recommend the limit of exactly 3 levels ([Chris Coyer](https://css-tricks.com/sass-style-guide/), [David Clark](http://davidtheclark.com/scss-lint-styleguide/#rule-set-innards), [Github](http://primercss.io/guidelines/#misc), [Hugo Giraudel](https://sass-guidelin.es/#selector-nesting), [jQuery](https://contribute.jquery.org/style-guide/css/#formatting), and so on)

I should also note, that in stylelint `max-nesting-depth: 3` [actually means "4"](http://stylelint.io/user-guide/rules/max-nesting-depth/), which gives one additional level of "freedom". So might also want to make it `2`.

And `ignore: ["at-rules-without-declaration-blocks"]` could be considered [to exclude @-rules without declarations from counting levels](http://stylelint.io/user-guide/rules/max-nesting-depth/#ignore-at-rules-without-declaration-blocks)